### PR TITLE
Extracted many callbacks to JabRefGUI and Globals, extracted Telemetry class

### PR DESCRIPTION
**Pull request by calixtus**
### Mandatory checks
- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

File: src/main/java/org/jabref/gui/externalfiles/DownloadFullTextAction.java [hunk 1]
```diff
@@ -102,23 +106,15 @@ protected Map<BibEntry, Optional<URL>> call() {
...
-        Globals.TASK_EXECUTOR.execute(findFullTextsTask);
+        taskExecutor.execute(findFullTextsTask);
     }
 
     private void downloadFullTexts(Map<BibEntry, Optional<URL>> downloads, BibDatabaseContext databaseContext) {
         for (Map.Entry<BibEntry, Optional<URL>> download : downloads.entrySet()) {
             BibEntry entry = download.getKey();
             Optional<URL> result = download.getValue();
             if (result.isPresent()) {
-                Optional<Path> dir = databaseContext.getFirstExistingFileDir(preferences.getFilePreferences());
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/gui/externalfiles/DownloadFullTextAction.java:113]: why did you remove this?

File: src/main/java/org/jabref/gui/fieldeditors/CitationKeyEditor.java [hunk 2]
```diff
@@ -20,41 +20,41 @@
...
-                             UndoManager undoManager,
-                             DialogService dialogService) {
+                             BibDatabaseContext databaseContext) {
+
+        ViewLoader.view(this)
+                  .root(this)
+                  .load();
 
-        this.preferences = preferences;
         this.viewModel = new CitationKeyEditorViewModel(
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/gui/fieldeditors/CitationKeyEditor.java:44]: we really need to make sure that this does not throw an NPE or such thing at runtime

File: src/main/java/org/jabref/gui/externalfiles/DownloadFullTextAction.java (hunk 1 above)
Comment (calixtus) [reply src/main/java/org/jabref/gui/externalfiles/DownloadFullTextAction.java:113]: It was not used anymore since a while, but not removed before.
LinkedFileViewModel::download determines itself where to download to.

File: src/main/java/org/jabref/gui/fieldeditors/CitationKeyEditor.java (hunk 2 above)
Comment (calixtus) [reply src/main/java/org/jabref/gui/fieldeditors/CitationKeyEditor.java:44]: I think I catched all the NPEs in the interface.
Happens only if the ViewLoader.load is called after the ViewModel is created, because DI is done by the ViewLoader. Must always be the first thing in the Constructor. Except if you use the FXML to reference stuff directly in the ViewModel. In that case you have to first call `Injector.registerExistingAndInject(this);` to do DI, then load the ViewModel, then call ViewLoader.load to parse the FXML

Comment (calixtus) [reply src/main/java/org/jabref/gui/fieldeditors/CitationKeyEditor.java:44]: See IdentifierEditor. Took me a bit, until I fixed that one.
